src/Connector.SqlServer/Connector/SqlServerConnector.cs

```diff
@@ -125,7 +125,7 @@ public Task<string> GetCorrelationId()
                             dataToUse, _defaultKeyFields, _logger);
 
                     foreach (var command in commands)
-                        await _client.ExecuteCommandAsync(config, command.Text, command.Parameters);
+                        await ExecuteCommandWithRetryAsync(_client.ExecuteCommandAsync(config, command.Text, command.Parameters));
```

Comment:
All of these needs to be changed to `ExecuteCommandWithRetryAsync(() => _client....)` otherwise you are retrying the an existing task that was already failed.

Reply:
Done